src/lib/support/ReadOnlyBuffer.h

```diff
@@ -48,12 +45,12 @@ class GenericAppendOnlyBuffer
     /// can be appended to the internal buffer;
     ///
     /// This will cause the internal buffer to become an allocated buffer
-    CHIP_ERROR EnsureAppendCapacity(size_t numElements);
+    [[nodiscard]] CHIP_ERROR EnsureAppendCapacity(size_t numElements);
```

Comment:
Why are the `[[nodiscard]]` needed?

Reply:
It seems to enforce people actually handling CHIP_ERROR returns. Not "needed" per se, but it seems like a great idea. Already have found places where we used these as they would be `(void)`. I would like people to check error codes and not drop them to the floor.